app/tests/tests_10/test_external_assets_endpoint.py

```diff
@@ -0,0 +1,238 @@
+
+        asset_data = asset.get_json('put')
+        asset_data['href'] = settings.EXTERNAL_TEST_ASSET_URL
+
+        response = self.client.put(
+            reverse_version('asset-detail', args=[collection.name, item.name, asset.attr_name]),
+            data=asset_data,
+            content_type='application/json'
+        )
+
+        json_data = response.json()
+        self.assertStatusCode(200, response)
+        self.assertEqual(json_data['href'], asset_data['href'])
+
+        updated_asset = Asset.objects.last()
+        self.assertEqual(updated_asset.file, asset_data['href'])
+        self.assertTrue(updated_asset.is_external)
+
+    def test_get_asset_with_external_url(self):
+        collection = self.collection
+        item = self.item
+
+        asset = self.factory.create_asset_sample(
+            item=self.item, sample='external-asset', db_create=True
+        )
+
+        response = self.client.get(
+            reverse_version('asset-detail', args=[collection.name, item.name, asset.attr_name])
+        )
+
+        json_data = response.json()
+
+        self.assertEqual(json_data['href'], asset.attr_file)
+
+    def test_create_asset_with_invalid_external_url(self):
+        collection = self.collection
+        item = self.item
+
+        collection.allow_external_assets = True
+        collection.save()
+
+        asset_data = {
+            'id': 'clouds.jpg',
+            'description': 'High in the sky',
+            'type': 'image/jpeg',  # specify the file explicitly
+            'href': 'this-is-not-a-url'
+        }
+
+        # create the asset
+        response = self.client.put(
+            reverse_version('asset-detail', args=[collection.name, item.name, asset_data['id']]),
+            data=asset_data,
+            content_type="application/json"
+        )
+
+        self.assertStatusCode(400, response)
+        description = response.json()['description']
+        self.assertIn('href', description, msg=f'Unexpected field error {description}')
+
+        self.assertEqual(
+            "Invalid URL provided", description['href'][0], msg="Unexpected error message"
+        )
+
+    def test_create_asset_with_inexistent_external_url(self):
+        collection = self.collection
+        item = self.item
+
+        collection.allow_external_assets = True
+        collection.save()
+
+        asset_data = {
+            'id': 'clouds.jpg',
+            'description': 'High in the sky',
+            'type': 'image/jpeg',  # specify the file explicitly
+            'href': settings.EXTERNAL_TEST_ASSET_URL
+        }
+
+        # create the asset with an existing one
+        response = self.client.put(
+            reverse_version('asset-detail', args=[collection.name, item.name, asset_data['id']]),
+            data=asset_data,
+            content_type="application/json"
+        )
+
+        self.assertStatusCode(201, response)
+
+        asset_data['href'] = 'https://bgdi.ch/notexist.jpg'
```

Comment:
maybe also use `test-domain.test` here (bgdi.ch actually exists and is forwarded to www.geo.admin.ch)

Reply:
The thing here is that I need something that can establish a connection but then returns 404, because the behaviour is different compared to the next test. But It should mention that, and I could easily use geo.admin.ch :+1: